src/withdrawals/tasks.py

```diff
@@ -63,76 +58,19 @@ async def _is_withdrawal_interval_passed(
-            return None
-        consensus_validators = await fetch_consensus_validators(
-            [event.public_key for event in events]
-        )
-        public_key_to_index = {val.public_key: val.index for val in consensus_validators}
-
-        for event in events[::-1]:  # reverse order to get the latest event
-            if await WithdrawalIntervalMixin.is_event_withdrawal_processed(
-                event=event,
-                current_slot=current_slot,
-                public_key_to_index=public_key_to_index,
-            ):
-                return event.block_number
+        if events:
+            return events[-1].block_number
 
         return None
 
-    @staticmethod
-    async def is_event_withdrawal_processed(
-        event: WithdrawalEvent, current_slot: int, public_key_to_index: dict[HexStr, int]
-    ) -> bool:
-        # pylint: disable=line-too-long
-        """
-        Check that the event withdrawal was successfully processed by the consensus layer.
-        - A request that passed the execution layer can still be reverted in the consensus layer
-        https://github.com/ethereum/consensus-specs/blob/dev/specs/electra/beacon-chain.md#new-process_withdrawal_request
-        - The request may also enter the execution layer's withdrawal queue before being processed by the consensus layer
-        https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7002.md#message-queue
-        """
-        event_slot = await calc_slot_by_block_number(event.block_number)
-
-        for withdrawal_slot in range(event_slot, current_slot + 1):
-            pending_partial_withdrawals = await consensus_client.get_pending_partial_withdrawals(
-                withdrawal_slot
-            )
-            # check if the event withdrawal is in the pending partial withdrawals
-            for withdrawal in pending_partial_withdrawals:
-                if (
-                    int(withdrawal['validator_index']) == public_key_to_index[event.public_key]
-                    and Web3.to_wei(withdrawal['amount'], 'gwei') == event.amount
-                ):
-                    return True
-
-            # Withdrawal queue can be full, check that request was added on this slot
-            # Otherwise check next slot
-            consensus_block = await consensus_client.get_block(str(withdrawal_slot))
-            execution_withdrawals = consensus_block['data']['message']['body'][
-                'execution_requests'
-            ]['withdrawals']
-            for request in execution_withdrawals:
-                # request was added but reverted by consensus layer
-                if (
-                    request['validator_pubkey'] == event.public_key
-                    and Web3.to_wei(request['amount'], 'gwei') == event.amount
-                ):
-                    return False
-
-        # event block is not finalized or withdrawal is still processing via execution client layer
-        raise LastWithdrawalNotProcessedError
```

Comment:
I’ve removed this part because, in a scenario where the transaction is submitted but rejected in consensus, it would increase the number of submissions and thus raise the operator’s gas costs. We want to avoid that in _is_withdrawal_interval_passed.